lib/Index/IndexRecord.cpp

```diff
@@ -513,14 +514,33 @@ emitDataForSwiftSerializedModule(ModuleDecl *module,
   if (*isUptodateOpt)
     return false;
 
+  // Reload resilient modules from swiftinterface to avoid indexing
```

Comment:
Is it possible to get the name of the reloaded module before actually loading it? We'll be using the `.swiftmodule` name now which is a bit weird, but maybe that's fine 🤔

Reply:
With the last change I think that's ok, we set aside `filename` before reloading.